src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -132,7 +132,12 @@ public override void OnStopActivity(Activity activity, object payload)
 
                 Status status = SpanHelper.ResolveSpanStatusForHttpStatusCode((int)response.StatusCode);
                 activity.AddTag(SpanAttributeConstants.StatusCodeKey, SpanHelper.GetCachedCanonicalCodeString(status.CanonicalCode));
-                activity.AddTag(SpanAttributeConstants.StatusDescriptionKey, response.HttpContext.Features.Get<IHttpResponseFeature>().ReasonPhrase);
+
+                var statusDescription = response.HttpContext.Features.Get<IHttpResponseFeature>().ReasonPhrase;
+                if (string.IsNullOrEmpty(statusDescription))
```

Comment:
Did you mean it to be `!string.IsNullOrEmpty(statusDescription)`?

Reply:
yes. Unfortunately the tests did not capture it. I was discussing with Eddy and realized asp.net core lacks same testing coverage as asp.net. Eddy is submitting a PR after this to add more coverage!